core/src/plugins/kubernetes/kubernetes-type/handlers.ts

```diff
@@ -59,6 +61,22 @@ export const kubernetesHandlers: Partial<ModuleActionHandlers<KubernetesModule>>
     const files = module.spec.files || []
     const manifests = module.spec.manifests || []
 
+    const apiVersion = ctx.projectApiVersion
+    let fileSources: KubernetesDeployActionSpecFileSources
+    if (apiVersion === GardenApiVersion.v2) {
+      fileSources = {
+        files: [],
+        manifestFiles: files,
+        manifestTemplates: [],
```

Comment:
If api version is v2, it still can contain template strings. We just don't want to use `legacyAllowPartial` anymore
Uh god, this is so confusing sorry for that :) 

```suggestion
        manifestFiles: [],
        manifestTemplates: files,
```

Reply:
Ah, sorry, a silly mistake :) FIxed in force-pushed commit 9ea77414e169959db7ef36374bb505ccedbac38d.